src/components/TeamGeneratorPage/TeamGenerator.js

```diff
@@ -1,52 +1,51 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useContext } from 'react';
 import ImportPlayerList from './ImportPlayerList';
 import TeamCompCard from './common/TeamCompCard';
 import TeamsDisplay from './TeamsDisplay';
-import { eventsNames } from '../../utils/consts';
 import TeamsHandler from './services/TeamsHandler';
+import { DisplayContext } from './services/Context';
 
 const TeamGenerator = () => {
 	const [playerList, setPlayerList] = useState([]);
+	const displayContext = useContext(DisplayContext);
 
 	const removePlayer = (player) => {
 		TeamsHandler.removePlayers(player);
-		setPlayerList(TeamsHandler.getAllPlayers());
+		setPlayerList(TeamsHandler.getPublicGroup());
 	};
 
 	function importPlayerList(importPlayers) {
 		TeamsHandler.addPlayers(importPlayers);
-		setPlayerList(TeamsHandler.getAllPlayers());
+		setPlayerList(TeamsHandler.getPublicGroup());
 	}
 
-	function addPlayerToPreTeam(e) {
-		const { player, team } = e.detail;
-		TeamsHandler.addPlayerToPreTeam(player, team);
-		setPlayerList(TeamsHandler.getAllPlayers());
-		window.dispatchEvent(new Event(eventsNames.updateTeams));
+	function loadPlayersFromStorage(params) {
+		TeamsHandler.restoreTeamsHandler();
+		setPlayerList(TeamsHandler.getPublicGroup());
+		displayContext.toUpdate.forEach((f) => f());
 	}
 
-	useEffect(() => {
-		const clearListener = window.addEventListener(eventsNames.clear, () => setPlayerList([]));
-		const addPreTeamListener = window.addEventListener(
-			eventsNames.addPlayerToPreTeam,
-			addPlayerToPreTeam,
-		);
+	function clear() {
+		setPlayerList([]);
+	}
 
-		TeamsHandler.restoreTeamsHandler();
-		setPlayerList(TeamsHandler.getAllPlayers());
-		window.dispatchEvent(new Event(eventsNames.updateTeams));
+	function updateDisplay() {
+		setPlayerList(TeamsHandler.getPublicGroup());
+	}
 
-		return () => {
-			window.removeEventListener(clearListener);
-			window.removeEventListener(addPreTeamListener);
-		};
+	useEffect(() => {
+		displayContext.toClear.push(clear);
+		displayContext.toUpdate.push(updateDisplay);
+		loadPlayersFromStorage();
 	}, []);
 
 	return (
 		<div className='flex flex-col items-center'>
-			<ImportPlayerList callback={importPlayerList}></ImportPlayerList>
-			<TeamCompCard list={playerList} name={'Players'} removeFunc={removePlayer}></TeamCompCard>
-			<TeamsDisplay playerList={playerList}></TeamsDisplay>
+			<DisplayContext.Provider value={displayContext}>
```

Comment:
if you wrap your main.js with a provider you don't have to do it here and your whole project will be able to use the useContext